lib/src/sv_sign.c

```diff
@@ -869,10 +871,13 @@ signed_video_set_private_key(signed_video_t *self, const char *private_key, size
     assert(certificate_chain);
     size_t certificate_chain_size = strlen(certificate_chain);
     status = port_settings_to_onvif(self);
-    if (status != SV_OK) return status;
-
-    return msrc_to_svrc(onvif_media_signing_set_signing_key_pair(self->onvif, private_key,
-        private_key_size, certificate_chain, certificate_chain_size, true));
+    if (status == SV_OK) {
+      return msrc_to_svrc(onvif_media_signing_set_signing_key_pair(self->onvif, private_key,
```

Comment:
What if this call fails, shouldn't you free `self->onvif` then as well? Otherwise you get different behaviors for different type of failures.